Mathlib/GroupTheory/GroupAction/SubMulAction/Combination.lean

```diff
@@ -0,0 +1,315 @@
+  exact Equiv.Perm.ext_iff.mpr hg
+
+theorem mulAction_faithful {G : Type*} [Group G] {α : Type*} [MulAction G α] {n : ℕ}
+    [DecidableEq α] (hn : 1 ≤ n) (hα : n < ENat.card α)
+    {g : G} :
+    MulAction.toPerm g = (1 : Equiv.Perm (n.Combination α))
+      ↔ MulAction.toPerm g = (1 : Equiv.Perm α) := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · contrapose h with h
+    have : ∃ a, (g • a : α) ≠ a := by simpa [Equiv.ext_iff] using h
+    obtain ⟨a, ha⟩ := this
+    obtain ⟨s, has, has'⟩ := exists_mem_notMem hn hα (Ne.symm ha)
+    rw [Equiv.ext_iff, not_forall]
+    use s
+    intro hs
+    apply has'
+    simp only [MulAction.toPerm_apply, Equiv.Perm.coe_one, id_eq] at hs
+    rw [← hs]
+    simpa only [coe_smul, smul_mem_smul_finset_iff, ← mem_coe_iff]
+  · ext1 s
+    simp only [MulAction.toPerm_apply, Equiv.Perm.coe_one, id_eq]
+    suffices ((g • s : n.Combination α) : Finset α) =
+      (MulAction.toPerm g : Equiv.Perm α) '' (s : Finset α) by
+      rw [← Subtype.coe_inj, ← Finset.coe_inj, this]
+      simp [h]
+    simp
+
+/-- If a group `G` acts faithfully on `α`,
+then it acts faithfull on `n.Combination α`,
+provided `1 ≤ n < ENat.card α`. -/
+theorem faithfulSMul [DecidableEq α] (hn : 1 ≤ n) (hα : n < ENat.card α)
+    [FaithfulSMul G α] :
+    FaithfulSMul G (n.Combination α) := by
+  rw [faithfulSMul_iff]
+  intro g hg
+  apply MulAction.toPerm_injective (α := G) (β := α)
+  rw [MulAction.toPerm_one, ← mulAction_faithful hn hα]
+  exact Equiv.Perm.ext_iff.mpr hg
+
+attribute [to_additive existing] faithfulSMul
+
+variable (α G)
+
+variable (n) in
+/-- The equivariant map from embeddings of `Fin n` (aka arrangement) to combinations. -/
+@[to_additive /-- The equivariant map from embeddings of `Fin n`
+  (aka arrangements) to combinations. -/]
+def mulActionHom_of_embedding [DecidableEq α] :
+    (Fin n ↪ α) →[G] n.Combination α where
+  toFun f := ⟨Finset.univ.map f, by rw [mem_iff, Finset.card_map, Finset.card_fin]⟩
+  map_smul' g f := by
+    simp only [id]
+    rw [← Subtype.coe_inj, Subtype.coe_mk, coe_smul,
+      f.smul_def, Finset.smul_finset_def,
+      ← Finset.map_map, Finset.map_eq_image]
+    simp [toPerm]
+
+@[to_additive]
+theorem coe_mulActionHom_of_embedding [DecidableEq α] (f : Fin n ↪ α) :
+    ↑((mulActionHom_of_embedding G α n).toFun f) = Finset.univ.map f :=
+  rfl
+
+@[to_additive]
+theorem mulActionHom_of_embedding_surjective [DecidableEq α] :
+    Function.Surjective (mulActionHom_of_embedding G α n) := by
+  intro ⟨s, hs⟩
+  obtain ⟨f : Fin n ↪ α, hf⟩ :=
+    Function.Embedding.exists_of_card_eq_finset (by rw [hs, Fintype.card_fin])
+  exact ⟨f, Subtype.ext hf⟩
+
+protected theorem card [Finite α] :
+    Nat.card (n.Combination α) = (Nat.card α).choose n := by
+  have := Fintype.ofFinite α
+  suffices n.Combination α = Finset.powersetCard n (Finset.univ : Finset α) by
+    simp [this]
+  ext; simp
+
+variable {α} in
+theorem _root_.ENat.nontrivial_combination (h1 : 0 < n) (h2 : n < ENat.card α) :
```

Comment:
Why not just `Nat.Combination.nontrivial`? Or `Nat.Combination.nontrivial_of_lt_enat_card` if you want that distinction? It just feels weird to put this in the `ENat` namespace.